### ISSUE TITLE
[GTK] Datepicker not working in Linux Wayland

### ISSUE THREAD

**Issue by Miepee** [picoe/Eto#2256]
In Linux Wayland, the Datepicker is not working

## Expected Behavior
A date picker pops up, letting me able to choose a date

## Actual Behavior
A small window appears for half a second, and then immediately disappears again, unable to let me choose a date.
The following message gets sent to STDOUT:
`Gdk-Message: 14:25:23.867: Window 0x55c42ef86240 is a temporary window without parent, application will not be able to position it on screen.`

## Steps to Reproduce the Problem

  1. Open Eto.Test.GTK in a Linux Wayland Session

## Specifications

  - Version: 2.7.0
  - Platform(s): Eto.GTK
  - Specs:
Operating System: Arch Linux
KDE Plasma Version: 5.25.3
Graphics Platform: Wayland
GTK Theme: Breeze

**Comment by cwensley** [comment]
Seems to be working with Ubuntu 22.04 running on Wayland.  Is this still an issue?
<img width="841" alt="Screenshot 2023-03-17 at 12 42 35 AM" src="https://user-images.githubusercontent.com/367446/225842952-3f2ace65-0915-4bff-a0f9-a81cdd1f6dc8.png">

**Comment by Miepee** [comment]
Yes, tested on opensuse tumbleweed with both GNOME and KDE on commit 25d9dd822b370464165fb2f895ba8edfcdd2aa42, its still an issue.
My first guesses are
- you aren't actually on wayland (should be able to confirm this quickly by just going to system settings -> info)
- 22.04 has some outdated packages where it works. Maybe someone could try 22.10 or the [23.04](https://cdimage.ubuntu.com/daily-live/current/) builds?
- VM vs real hardware makes a difference for some reason

**Comment by cobalt2727** [comment]
I'm able to replicate this, confirmed to happen on Pop!_OS 22.04 (which, like Ubuntu 22.04, has Gnome 42.5)

@cwensley what is the output of `echo $XDG_SESSION_TYPE` when you throw that in a terminal?
